source/concurrent/Thread.ooc

```diff
@@ -12,19 +12,39 @@ import native/[ThreadUnix, ThreadWin32]
 
 Thread: abstract class {
 	_code: Func
+	_canceled := false
+	_cancelable: Bool
+	_cancelMutex: Mutex = null
+	_ownsCancelMutex: Bool
```

Comment:
I'm still not sure if we shouldn't move all of this to a separate `Task` class, this seems like quite a lot of data members to add in `Thread`. This way Thread could accept a Task object, which wraps the cancelling / mutex ownership logic internally. Maybe another PR later ?